roles/identity-management/populate-users/tasks/main.yml

```diff
@@ -2,10 +2,22 @@
 - name: Convert csv to json - set facts
   delegate_to: localhost
   set_fact:
-    identities:
+    csv_identities:
       users: "{{ lookup('csvtojson', 'file=' + csv_doc_file_name + ' var=users') }}"
       groups: "{{ lookup('csvtojson', 'file=' + csv_doc_file_name + ' var=user_groups') }}"
 
+- name: Merge imported identities list with static identities list
+  delegate_to: localhost
+  set_fact:
+    merged_identities:
+      users: "{{ identities.users|default([]) + csv_identities.users }}"
+      groups: "{{ identities.groups|default([]) + csv_identities.groups }}"
+
+- name: Combine identities dict with merged user and group identities
+  delegate_to: localhost
+  set_fact:
+    identities: "{{ identities|default({}) | combine(merged_identities) }}"
```

Comment:
I believe you'd want to add the ` recursive=True` flag to the combine filter to ensure it all gets merged. Otherwise it will just combine at the top level and you may loose some important content, no?